enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/LdapRealm.java

```diff
@@ -448,28 +463,31 @@ private String parseLdapServerUrl( String rawLdapServer )
         return roleNames;
     }
 
-    private boolean validateUserSearchSettings()
+    private void assertValidUserSearchSettings()
     {
         boolean proceedWithSearch = true;
 
         if ( userSearchBase == null || userSearchBase.isEmpty() )
         {
-            log.warn( "LDAP user search base is empty." );
+            securityLog.warn( "LDAP user search base is empty." );
             proceedWithSearch = false;
         }
         if ( userSearchFilter == null || !userSearchFilter.contains( "{0}" ) )
         {
-            log.warn( "LDAP user search filter does not contain the argument placeholder {0}, so the search result " +
-                      "will be independent of the user principal." );
+            securityLog.warn( "LDAP user search filter does not contain the argument placeholder {0}, " +
+                    "so the search result will be independent of the user principal." );
         }
         if ( membershipAttributeNames == null || membershipAttributeNames.isEmpty() )
         {
             // If we don't have any attributes to look for we will never find anything
-            log.warn( "LDAP group membership attribute names are empty. Authorization will not be possible." );
+            securityLog.warn( "LDAP group membership attribute names are empty. Authorization will not be possible." );
             proceedWithSearch = false;
         }
 
-        return proceedWithSearch;
+        if ( !proceedWithSearch )
+        {
+            throw new IllegalArgumentException( "Illegal LDAP user search settings, see security log for details." );
```

Comment:
If we now throw exception then maybe we should change warnings to errors for those cases